squigglepy/utils.py

```diff
@@ -703,6 +703,71 @@ def laplace(s, n=None, time_passed=None,
         raise ValueError('Fatal logic error - programmer made mistake!')
 
 
+def growth_rate_to_doubling_time(growth_rate):
+    """
+    Convert a positive growth rate (expressed as a number, numpy array or distribution
+    where 0.05 means +5%) to a doubling time. The time unit remains the same, so if we've
+    got +5% annual growth, the returned value is the doubling time in years.
+
+    NOTE: This only works works for numbers, arrays and distributions where all numbers
+    are above 0. (Otherwise it makes no sense to talk about doubling times.)
+
+    Parameters
+    ----------
+    growth_rate : float or np.array or BaseDistribution
+        The growth rate expressed as a fraction (the percentage divided by 100).
+
+    Returns
+    -------
+    float or np.array or ComplexDistribution
+        Returns the doubling time.
+
+    Examples
+    --------
+    >>> growth_rate_to_doubling_time(0.01)
+    69.66071689357483
+    """
+    if is_dist(growth_rate):
+        return math.log(2) / (1.0 + growth_rate).log()
+    elif isinstance(growth_rate, np.ndarray):
```

Comment:
Unsure but you could consider use `_is_numpy` from utils if you want to capture numpy things other than numpy arrays